### PR TITLE
Security check

### PR DESCRIPTION
**Pull request by TheFakeElon**
[Base Branch]: master

[Release Notes]: fixes a security flaw in tgs3
tgstation/tgstation#62568
:cl:
fixes a security flaw
/:cl:

### CODE REVIEW

**Review by ExcessiveUseOfCobblestone — APPROVED** [review]

**Comment by LemonInTheDark** [comment]
Mothblocks is blind #1349

**Comment by LemonInTheDark** [comment]
wait wrong version

**Comment by MCterra10** [comment]
can we merge this already?

**Comment by LemonInTheDark** [comment]
no, because it breaks servers that don't have that proc (Insert better reason cyber will give here)
This whole thing is so mindnummingly dumb I wish to return to an era before security posting

File: src/DMAPI/tgs/v3210/api.dm [hunk 1]
```diff
@@ -54,6 +54,8 @@
 	return ""
 
 /datum/tgs_api/v3210/proc/file2list(filename)
+	if(IsAdminAdvancedProcCall())
+		CRASH("Attempted to read file via admin call")
```
Comment (alexkar598) [inline src/DMAPI/tgs/v3210/api.dm:58]: This proc is not part of DMAPI

**Review by alexkar598 — CHANGES_REQUESTED** [review]

**Comment by AffectedArc07** [comment]
The real fix here is to stop using deprecated software which was coded in one afternoon while cyberboss was stoned
image

**Comment by AffectedArc07** [comment]
Putting another post. You can mitigate this on your own codebases outside of the DMAPI. Here are a few ways in no particular order:
- `GENERAL_PROTECT_DATUM(/datum/tgs_api/v3210)`
- Defining a proc after this to make inheritence work. Just put this in your codebase after your DMAPI folder
```
/datum/tgs_api/v3210/file2list(filename)
	if(IsAdminAdvancedProcCall())
		CRASH("Attempted to read file via admin call")
	..()
```

**Comment by Cyberboss** [comment]
Various reasons/alternative already stated, plus you could always redefine file2text. Modifying external library code is not the wei here.

**Comment by Cyberboss** [comment]
Revisionist history 
image

**Pull request closed without merging**